tests/contrib/io/cached/test_cached_dataset.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2018-2019 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited (“QuantumBlack”) name and logo
+# (either separately or in combination, “QuantumBlack Trademarks”) are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=protected-access
+import pickle
+from io import StringIO
+from pickle import PicklingError
+
+import pytest
+import yaml
+
+from kedro.contrib.io.cached import CachedDataSet
+from kedro.io import CSVLocalDataSet, DataCatalog, DataSetError, MemoryDataSet
+
+YML_CONFIG = """
+test_ds:
+  type: kedro.contrib.io.cached.CachedDataSet
+  dataset:
+      type: CSVLocalDataSet
+      filepath: example.csv
+"""
+
+YML_CONFIG_VERSIONED = """
+test_ds:
+  type: kedro.contrib.io.cached.CachedDataSet
+  versioned: true
+  dataset:
+      type: CSVLocalDataSet
+      filepath: example.csv
+"""
+
+YML_CONFIG_VERSIONED_BAD = """
+test_ds:
+  type: kedro.contrib.io.cached.CachedDataSet
+  dataset:
+      type: CSVLocalDataSet
+      filepath: example.csv
+      versioned: true
+"""
+
+
+@pytest.fixture
+def cached_ds():
+    wrapped = MemoryDataSet()
+    return CachedDataSet(wrapped)
+
+
+class TestCachedDataset:
+    def test_load_empty(self, cached_ds):
+        with pytest.raises(DataSetError, match=r"has not been saved yet"):
+            _ = cached_ds.load()
```

Comment:
@tsanikgr Was reading the PR just out of curiosity--what's the purpose of assigning to `_` across your tests? I assume it's because the most frequent usage is to assign to a variable, but that's neither required nor always going to be the case.